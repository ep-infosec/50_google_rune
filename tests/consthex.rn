//  Copyright 2021 Google LLC.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

s = "0123456789aBcDeF"
for i in range(s.length()) {
  c = s[i]
  println c, " = ", (9 & !-(c >> 6)) + (c & 0xf)
}
for val = 0u8, val < 16, val += 1 {
  bit1 = val >> 1
  bit2 = val >> 2
  bit3 = val >> 3
  delta = "a"[0] - "0"[0] - 10
  s = "0"
  s[0] = (!-((bit1 | bit2) & bit3) & delta) + "0"[0] + val
  println s
}
