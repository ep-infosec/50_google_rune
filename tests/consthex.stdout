48 = 0
49 = 1
50 = 2
51 = 3
52 = 4
53 = 5
54 = 6
55 = 7
56 = 8
57 = 9
97 = 10
66 = 11
99 = 12
68 = 13
101 = 14
70 = 15
0
1
2
3
4
5
6
7
8
9
a
b
c
d
e
f
